zrml/prediction-markets/src/lib.rs

```diff
@@ -2688,6 +2724,15 @@ mod pallet {
             Ok(())
         }
 
+        fn ensure_creator_of_trusted_market(
+            market: &MarketOf<T>,
+            who: &T::AccountId,
+        ) -> DispatchResult {
+            ensure!(market.creator == *who, Error::<T>::CallerNotMarketCreator);
+            ensure!(market.dispute_mechanism.is_none(), Error::<T>::MarketIsNotTrusted);
+            Ok(())
+        }
+
```

Comment:
Why are we abstracting this into a separate function? Seems like a very specific thing to verify.

Reply:
4f591cf0660d9d0c836957aeead58e1ddebcf97b